### PR TITLE
*: update client-go to improve availability on tikv failure

### LINKED ISSUE #34906: Recover time may be very long if TiKV enables hibernate region
## Enhancement

Considering one TiKV is unhealthy, all requests sending to that store will timeout. But due to hibernate region, the regions whose leaders are on the unhealthy store will not be re-elected until a request is sent to the follower.

If there are thousands of such regions, every time we are accessing a new region, it will send a request to the original leader on the unhealthy TiKV because the leader definitely does not change. And then, the user will always experience very long request duration until all regions are touched and elect new leaders.

We can use the health check service of TiKV to get the serving status of TiKV. By filtering out healthy nodes, we can recover from TiKV failure more quickly.

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve?

Issue Number: close #34906

### What is changed and how it works?

See https://github.com/tikv/client-go/pull/503

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Prevent sending requests to unhealthy TiKV nodes to improve availability.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -63,7 +63,7 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.2-0.20220504104629-106ec21d14df
 	github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2
-	github.com/tikv/client-go/v2 v2.0.1-0.20220531092439-efebaeb9fe53
```
Comment (hawkingrei) [inline go.mod:66]: Don't forget to update bazel's dependencies.

**Comment by cfzjywxk** [comment]
@sticnarf 
It seems the client-go does not have a corresponding release branch for v6.1?

File: go.mod (hunk 1 above)
Comment (sticnarf) [reply go.mod:66]: Release 6.1 doesn't seem to have bazel build files.

**Comment by sticnarf** [comment]
> @sticnarf It seems the client-go does not have a corresponding release branch for v6.1?

It does. Now it points to 1693f8c715244dd6aa713a49a47bcdfcfd87fa5b at https://github.com/tikv/client-go/commits/tidb-6.1

**Review by cfzjywxk — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (hawkingrei) [reply go.mod:66]: Sorry, I don't care about it.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**